### PR TITLE
Add AsyncHttp4sServlet test suite

### PR DESCRIPTION
**Pull request by vasilmkd**
I cannot seem to reproduce #2354.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -365,10 +365,11 @@ lazy val servlet = libraryProject("servlet")
     libraryDependencies ++= Seq(
       javaxServletApi % Provided,
       jettyServer % Test,
-      jettyServlet % Test,
+      jettyServlet % Test
     ),
   )
   .dependsOn(server % "compile;test->test")
+  .dependsOn(asyncHttpClient % Test)
```
Comment (rossabaker) [inline build.sbt:372]: This dependency is going to be a pain as we test across repos after the Great Schism.  We get close to circular dependencies when we test our servers with our clients.  How hard would it be to do this with raw AHC?

File: servlet/src/test/scala/org/http4s/servlet/AsyncHttp4sServletSuite.scala [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  val servletServer = ResourceFixture[Int](serverPortR)
+
+  def get(serverPort: Int, path: String): IO[String] =
+    testBlocker.delay[IO, String](
+      Source
+        .fromURL(new URL(s"http://127.0.0.1:$serverPort/$path"))
+        .getLines()
+        .mkString)
+
+  servletServer.test("Http4sBlockingServlet handle GET requests") { server =>
```
Comment (rossabaker) [inline servlet/src/test/scala/org/http4s/servlet/AsyncHttp4sServletSuite.scala:58]: The name seems off here.  Isn't it the servlet with NonBlockingIO?

**Review by rossabaker — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 1 above)
Comment (vasilmkd) [reply build.sbt:372]: I will do it.

File: servlet/src/test/scala/org/http4s/servlet/AsyncHttp4sServletSuite.scala (hunk 2 above)
Comment (vasilmkd) [reply servlet/src/test/scala/org/http4s/servlet/AsyncHttp4sServletSuite.scala:58]: Sorry, yeah, I guess I forgot to adjust some things when I originally copied the file contents.

Comment (vasilmkd) [reply servlet/src/test/scala/org/http4s/servlet/AsyncHttp4sServletSuite.scala]: Done.

File: build.sbt (hunk 1 above)
Comment (vasilmkd) [reply build.sbt]: Done.

**Review by rossabaker — APPROVED** [review]

**Pull request merged**